File: solutions/clojure/cars-assemble/1/src/cars_assemble.clj
(ns cars-assemble)

(defn production-rate
  "Returns the assembly line's production rate per hour,
   taking into account its success rate"
  [speed]
  (def rate 221)
  (cond (=  speed 0)  (0)
        (<= speed 4)  (100)
        (<= speed 8)  (0.90)
        (<= speed 9)  (0.80)
        (<= speed 10) (0.77)))

(defn working-items
  "Calculates how many working cars are produced per minute"
  [speed]
  (* (production-rate speed) 221))

File: solutions/go/cars-assemble/1/cars_assemble.go
package cars

// CalculateWorkingCarsPerHour calculates how many working cars are
// produced by the assembly line every hour.
func CalculateWorkingCarsPerHour(productionRate int, successRate float64) float64 {
	return (float64(productionRate) * successRate / 100)
}

// CalculateWorkingCarsPerMinute calculates how many working cars are
// produced by the assembly line every minute.
func CalculateWorkingCarsPerMinute(productionRate int, successRate float64) int {
	return int(float64(productionRate) * successRate / 100 / 60)
}

// CalculateCost works out the cost of producing the given number of cars.
func CalculateCost(carsCount int) uint {
	const costOfOne = 10000
        const costOfTen = 95000
        countsInTen := carsCount / 10
        countsInOne := carsCount % 10
        return uint( (countsInTen * costOfTen) + (countsInOne * costOfOne))
}
